### PR TITLE
Add EF Code First for social network

### PR DESCRIPTION
**Pull request by YevheniiTk**

### CODE REVIEW

File: SocialNetwork/App.config [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      <parameters>
+        <parameter value="mssqllocaldb" />
+      </parameters>
+    </defaultConnectionFactory>
+    <providers>
+      <provider invariantName="System.Data.SqlClient" type="System.Data.Entity.SqlServer.SqlProviderServices, EntityFramework.SqlServer" />
+    </providers>
+  </entityFramework>
+  <connectionStrings>
+    <add name="SocialNetworkContext" connectionString="data source=DESKTOP-4CBIOUT\MYMSSQLSERVER;initial catalog=SocialNetworkDB;integrated security=True;MultipleActiveResultSets=True;App=EntityFramework" providerName="System.Data.SqlClient" />
```
Comment (vmedvedovskiy) [inline SocialNetwork/App.config]: Let's move connectionStrings to external file and add this file to .gitignore. Everyone will have different value here, and so we will avoid overwriting this value

File: SocialNetwork/DataContext/Comment.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace SocialNetwork.DataContext
+{
+    public class Comment
+    {
+        [Key()]
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Comment.cs]: Braces can be omitted

File: SocialNetwork/DataContext/Comment.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        public Guid CommentId { get; private set; }
+
+
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Comment.cs]: Extra empty lines

File: SocialNetwork/DataContext/Comment.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        public DateTime CreateDate { get; set; }
+                
+        public string Content { get; set; }
+        
+        [ForeignKey("CommentId")]
+        public ICollection<Like> Likes { get; set; }
+
+        public Comment()
+        {
+            this.CommentId = Guid.NewGuid();
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Comment.cs]: Let's use property initializer from C# 6

File: SocialNetwork/DataContext/Content.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace SocialNetwork.DataContext
+{
+    public class Content
+    {
+        [Key]
+        public int TypeContent { get; set; }
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Content.cs]: ContentType

File: SocialNetwork/DataContext/Country.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace SocialNetwork.DataContext
+{
+    public class Country
+    {
+        [Key]
+        [MaxLength(3)]
+        public string Alpha3 { get; set; }
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Country.cs]: Alpha-CountryIsoCode

File: SocialNetwork/DataContext/Country.cs [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+        [MaxLength(2)]
+        public string Alpha2 { get; set; }
+
+        public string Name { get; set; }
+
+        public int NumericCode { get; set; }
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Country.cs]: What is a purpose of this field?

File: SocialNetwork/DataContext/Country.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+        public ICollection<User> Users { get; set; }
+
+        public Country(string name, string alpha2, string alpha3, int numericCode)
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Country.cs]: Do we need this constructor?

File: SocialNetwork/DataContext/Friend.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace SocialNetwork.DataContext
+{
+    public class Friend
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Friend.cs]: Friend -> UserFriend

File: SocialNetwork/DataContext/Like.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace SocialNetwork.DataContext
+{
+    public class Like
+    {
+        public Guid PostId { get; set; }
+        public Guid UserId { get; set; }
+        public Guid CommentId { get; set; }
+       
+        public Post Post { get; set; }
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Like.cs]: Let's declare this properties as virtual

File: SocialNetwork/DataContext/Post.cs [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace SocialNetwork.DataContext
+{
+    public class Post
+    {
+        [Key()]
+        public Guid PostId { get; private set; }
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Post.cs]: Empty line between properties

File: SocialNetwork/DataContext/Post.cs [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+        public User User { get; set; }
+        
+        public Guid AuthorId { get; set; }
+
+        public string Titl { get; set; }
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Post.cs]: Title

File: SocialNetwork/DataContext/Post.cs [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+        [Column(TypeName = "datetime2")]
+        public DateTime CreateDate { get; set; }
+
+        [Column(TypeName = "datetime2")]
+        public DateTime PublicationData { get; set; }
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/Post.cs]: Data->Date

File: SocialNetwork/DataContext/SocialNetworkContext.cs [hunk 14]
```diff
@@ -0,0 +1,39 @@
+namespace SocialNetwork.DataContext
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/SocialNetworkContext.cs]: Let's put project to C-18-01-Capstone.Main and rename it appropriately

File: SocialNetwork/DataContext/User.cs [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace SocialNetwork.DataContext
+{
+    public class User
+    {
+        [Key]
+        public Guid UserId { get; private  set; }
+        public string FName { get; set; }
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/User.cs]: FirstName

File: SocialNetwork/DataContext/User.cs [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+
+        public string LName { get; set; }
```
Comment (vmedvedovskiy) [inline SocialNetwork/DataContext/User.cs]: LastName

**Review by vmedvedovskiy — CHANGES_REQUESTED** [review]

**Comment by YevheniiTk** [comment]
added changes

**Review by vmedvedovskiy — APPROVED** [review]

**Pull request merged**